### PR TITLE
Admin role

### PR DESCRIPTION
**Pull request by darkBasic666**
## Summary
- Add role atribute in user model
- Add "/admin/users"  POST end point.
- Add user validation middleware for new endpoint
- Add check admin role middleware and use for new endpoint
- Test for new endpoint  
## Trello Card
https://trello.com/c/3gchyHiB/18-usuario-administrador

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -3,31 +3,63 @@ const bcrypt = require('bcrypt'),
...
-      user.password = hash;
+exports.createAdmin = (req, res, next) => {
+  const saltRounds = 10;
+  bcrypt.genSalt(saltRounds).then((verificationCode) => {
+    const user = buildUser(req, verificationCode);
 
-      userService.create(user).then((u) => {
+    userService.getByEmail(user.email).then((u) => {
+      if (u) {
+        u.role = 'admin';
```
Comment (igna92ts) [inline app/controllers/users.js]: I would rather have this role at least in a constant at the top of the file

File: app/controllers/users.js [hunk 2]
```diff
@@ -3,31 +3,63 @@ const bcrypt = require('bcrypt'),
   sessionManager = require('./../services/sessionManager'),
   userService = require('../services/users');
 
+const save = (user, res, next) => {
```
Comment (igna92ts) [inline app/controllers/users.js]: I'd rather have this function be called create or something along those lines. I think this name is a little unclear and sound more like an update that a creation.

File: app/middlewares/auth.js [hunk 3]
```diff
@@ -27,3 +27,13 @@ exports.secure = (req, res, next) => {
     res.end();
   }
 };
+
+exports.needAdminRole = (req, res, next) => {
+  const user = req.user;
+  if (user.role === 'admin') {
```
Comment (igna92ts) [inline app/middlewares/auth.js]: This is a possible security breach. I would prefer if you didnt send the role in the token and check it with the db directly when trying to do an admin only request

Comment (darkBasic666) [inline app/middlewares/auth.js]: Actually getting user using id to token user.

File: app/controllers/users.js (hunk 2 above)
Comment (darkBasic666) [inline app/controllers/users.js]: Done

File: app/controllers/users.js (hunk 1 above)
Comment (darkBasic666) [inline app/controllers/users.js]: Done

**Pull request merged**